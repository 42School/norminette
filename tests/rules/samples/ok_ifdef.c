#ifdef SF_DEBUG_MALLOC
void	*sf_malloc_impl_debug(sf_ulong size)
#else
void	*sf_malloc_impl(sf_ulong size, sf_malloc_type_t type)
#endif
{
	return (0);
}
