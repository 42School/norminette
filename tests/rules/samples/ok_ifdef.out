[36mok_ifdef.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 1":
		<IFDEF=#ifdef SF_DEBUG_MALLOC> <NEWLINE>
[36mok_ifdef.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 2":
		<VOID> <TAB> <MULT> <IDENTIFIER=sf_malloc_impl_debug> <LPARENTHESIS> <IDENTIFIER=sf_ulong> <SPACE> <IDENTIFIER=size> <RPARENTHESIS> <NEWLINE>
[36mok_ifdef.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 3":
		<#ELSE=#else> <NEWLINE>
[36mok_ifdef.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 4":
		<VOID> <TAB> <MULT> <IDENTIFIER=sf_malloc_impl> <LPARENTHESIS> <IDENTIFIER=sf_ulong> <SPACE> <IDENTIFIER=size> <COMMA> <SPACE> <IDENTIFIER=sf_malloc_type_t> <SPACE> <IDENTIFIER=type> <RPARENTHESIS> <NEWLINE>
[36mok_ifdef.c[0m - [32mIsPreprocessorStatement[0m In "Function" from "GlobalScope" line 5":
		<ENDIF=#endif> <NEWLINE>
[36mok_ifdef.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 6":
		<LBRACE> <NEWLINE>
[36mok_ifdef.c[0m - [32mIsExpressionStatement[0m In "Function" from "GlobalScope" line 7":
		<TAB> <RETURN> <SPACE> <LPARENTHESIS> <CONSTANT=0> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_ifdef.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 8":
		<RBRACE> <NEWLINE>
ok_ifdef.c: Error!
Error: INVALID_HEADER       (line:   1, col:   1):	Missing or invalid 42 header
Error: NL_AFTER_PREPROC     (line:   2, col:   1):	Preprocessor statement must be followed by a newline
Error: NL_AFTER_PREPROC     (line:   4, col:   1):	Preprocessor statement must be followed by a newline
Error: NL_AFTER_PREPROC     (line:   6, col:   1):	Preprocessor statement must be followed by a newline
